### PR TITLE
Make MONO_API_DATA clearer.

### PR DESCRIPTION
**Pull request by jaykrell**
Make MONO_API_DATA clearer, its old form -- MONO_API extern.
extern "C" { } should be added around it not within it, for clarity.
I've always found extern "C" without braces unclear, esp. for data.

i.e.

```
 extern "C" {
1   extern void function(); // redundant but clear
                            // same as void function();
2   extern int data; // not redundant, and clear
}
```

vs.
```
3   extern "C" void function(); // I guess clear.
4   extern "C" int data; // unclear -- is it extern or extern "C" or both?
```

This PR in particular turns 4 into 2.
2 is clearly a declaration, not a definition.
4 is unclear as to if it is a declaration or definition.

I was looking into https://github.com/mono/mono/issues/18827.

This does not actually change anything but makes things clearer.

This is similar/related in spirit to https://github.com/mono/mono/pull/18891.
It should not make a difference.
In this case, `#if __cplusplus` remains, but the meaning of it is clearer.
It becomes *only* about the common `extern "C" { }` and not even the less clear `extern "C"`.

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/mini-native-types.c [hunk 1]
```diff
@@ -42,7 +42,6 @@ typedef struct {
 	short compare_op;
 } MagicTypeInfo;
 
-
```
Comment (lambdageek) [inline mono/mini/mini-native-types.c]: unrelated whitespace change

File: mono/mini/mini-native-types.c [hunk 2]
```diff
@@ -175,7 +174,7 @@ emit_widen (MonoCompile *cfg, const MagicTypeInfo *info, int sreg)
 		MONO_INST_NEW (cfg, ins, info->conv_4_to_8);
 	ins->sreg1 = sreg;
 	ins->type = info->big_stack_type;
-	ins->dreg = alloc_dreg (cfg, info->big_stack_type); 
+	ins->dreg = alloc_dreg (cfg, info->big_stack_type);
```
Comment (lambdageek) [inline mono/mini/mini-native-types.c]: unrelated whitespace change

File: mono/utils/mono-publib.h [hunk 3]
```diff
@@ -75,15 +75,8 @@ typedef unsigned __int64	uint64_t;
...
-// extern "C" extern int c; // warning: duplicate 'extern' declaration specifier [-Wduplicate-decl-specifier]
-//
-// Therefore, remove extern on functions as always meaningless/redundant,
-// and provide MONO_API_DATA for data, that always has one and only one extern.
-#ifdef __cplusplus
-#define MONO_API_DATA MONO_API
-#else
-#define MONO_API_DATA extern MONO_API
-#endif
+// Should but not must wrap in extern "C" (MONO_BEGIN_DECLS, MONO_END_DECLS).
```
Comment (lambdageek) [inline mono/utils/mono-publib.h]:
```suggestion
// Should (but not must) wrap in extern "C" (MONO_BEGIN_DECLS, MONO_END_DECLS).
```

File: mono/mini/mini-native-types.c (hunk 1 above)
Comment (jaykrell) [reply mono/mini/mini-native-types.c]: That's already in another approved PR, I can remove it here.
It would be nice if we could actually cleanup the code formatting.

File: mono/mini/mini-native-types.c (hunk 2 above)
Comment (jaykrell) [reply mono/mini/mini-native-types.c]: Ditto.

File: mono/utils/mono-publib.h (hunk 3 above)
Comment (jaykrell) [reply mono/utils/mono-publib.h]: I wonder if I should clarify further.
In particular, if the consumers/produces are all in the same language (C or C++) then need not.
If any of the consumers/producers are C, then must.
